### PR TITLE
[BUGFIX release] Update `Ember.deprecate` and `Ember.warn` calls to include required information

### PR DESCRIPTION
**Pull request by HeroicEric**
fixes https://github.com/emberjs/data/issues/3568 for release branch

Related: https://github.com/emberjs/data/pull/3570

### CODE REVIEW

**Comment by HeroicEric** [comment]
@wecc this has all of the stuff from https://github.com/emberjs/data/pull/3570 as well as anything additional that was left in the release branch.

I assumed that everything that was still in the release branch but was no longer in master could be marked as `until: '2.0.0'`. Everything else is the same as it was in the other PR, so there may be some additional things that should be changed from `'3.0.0'` to `'2.0.0'`.

File: packages/ember-data/lib/main.js [hunk 1]
```diff
@@ -201,7 +205,10 @@ if (Ember.platform.hasPropertyAccessors) {
   Ember.defineProperty(DS, 'FixtureAdapter', {
     get: function() {
       if (_FixtureAdapter === FixtureAdapter) {
-        Ember.deprecate('DS.FixtureAdapter has been deprecated and moved into an unsupported addon: https://github.com/emberjs/ember-data-fixture-adapter/tree/master');
+        Ember.deprecate('DS.FixtureAdapter has been deprecated and moved into an unsupported addon: https://github.com/emberjs/ember-data-fixture-adapter/tree/master', false, {
+          id: 'ds.adapter.fixture-adapter-deprecated',
+          until: '3.0.0'
```
Comment (HeroicEric) [inline packages/ember-data/lib/main.js]: should be 2.0.0?

File: packages/ember-data/lib/system/adapter.js [hunk 2]
```diff
@@ -483,7 +483,7 @@ var Adapter = Ember.Object.extend({
   */
   shouldReloadAll: function(store, snapshotRecordArray) {
     var modelName = snapshotRecordArray.type.modelName;
-    Ember.deprecate(`The default behavior of shouldReloadAll will change in Ember Data 2.0 to always return false when there is at least one "${modelName}" record in the store. If you would like to preserve the current behavior please override shouldReloadAll in your adapter:application and return true.`);
+    Ember.deprecate(`The default behavior of shouldReloadAll will change in Ember Data 2.0 to always return false when there is at least one "${modelName}" record in the store. If you would like to preserve the current behavior please override shouldReloadAll in your adapter:application and return true.`, false, { id: 'ds.adapter.should-reload-all-default-behavior', until: '3.0.0' });
```
Comment (HeroicEric) [inline packages/ember-data/lib/system/adapter.js]: should be 2.0.0?

File: packages/ember-data/lib/system/adapter.js [hunk 3]
```diff
@@ -504,7 +504,7 @@ var Adapter = Ember.Object.extend({
     @return {Boolean}
   */
   shouldBackgroundReloadRecord: function(store, snapshot) {
-    Ember.deprecate('The default behavior of `shouldBackgroundReloadRecord` will change in Ember Data 2.0 to always return true. If you would like to preserve the current behavior please override `shouldBackgroundReloadRecord` in your adapter:application and return false.');
+    Ember.deprecate('The default behavior of `shouldBackgroundReloadRecord` will change in Ember Data 2.0 to always return true. If you would like to preserve the current behavior please override `shouldBackgroundReloadRecord` in your adapter:application and return false.', false, { id: 'ds.adapter.should-background-reload-record-default-behavior', until: '3.0.0' });
```
Comment (HeroicEric) [inline packages/ember-data/lib/system/adapter.js]: should be 2.0.0?

Comment (bmac) [inline packages/ember-data/lib/system/adapter.js]: yes

File: packages/ember-data/lib/system/store.js [hunk 4]
```diff
@@ -1745,7 +1783,10 @@ Store = Service.extend({
         enumerable: true,
         configurable: false,
         get: function() {
-          Ember.deprecate('Usage of `typeKey` has been deprecated and will be removed in Ember Data 1.0. It has been replaced by `modelName` on the model class.');
+          Ember.deprecate('Usage of `typeKey` has been deprecated and will be removed in Ember Data 1.0. It has been replaced by `modelName` on the model class.', false, {
```
Comment (HeroicEric) [inline packages/ember-data/lib/system/store.js]: should the deprecation message itself here change to say 2.0?

File: packages/ember-data/lib/system/adapter.js (hunk 2 above)
Comment (bmac) [inline packages/ember-data/lib/system/adapter.js]: yes

File: packages/ember-data/lib/main.js (hunk 1 above)
Comment (bmac) [inline packages/ember-data/lib/main.js]: yes

File: packages/ember-data/lib/system/store.js (hunk 4 above)
Comment (bmac) [inline packages/ember-data/lib/system/store.js]: :+1:

**Comment by HeroicEric** [comment]
after talking to @bmac in slack, all deprecations/warnings are now marked as until 2.0.0

File: packages/ember-data/lib/serializers/rest-serializer.js [hunk 5]
```diff
@@ -453,7 +462,10 @@ var RESTSerializer = JSONSerializer.extend({
       var modelName = this.modelNameFromPayloadKey(prop);
 
       if (!store.modelFactoryFor(modelName)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false);
+        Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false, {
+          id: 'ds.serializer.model-for-key-missing',
+          until: '3.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/serializers/rest-serializer.js]: Do we need `until` for `warn`s?

File: packages/ember-data/lib/serializers/rest-serializer.js [hunk 6]
```diff
@@ -617,7 +632,10 @@ var RESTSerializer = JSONSerializer.extend({
 
       var typeName = this.modelNameFromPayloadKey(modelName);
       if (!store.modelFactoryFor(typeName)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
+        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false, {
+          id: 'ds.serializer.model-for-key-missing',
+          until: '3.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/serializers/rest-serializer.js]: ...same here.

File: packages/ember-data/lib/serializers/rest-serializer.js [hunk 7]
```diff
@@ -684,13 +702,19 @@ var RESTSerializer = JSONSerializer.extend({
...
+    });
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
       var modelName = this.modelNameFromPayloadKey(prop);
       if (!store.modelFactoryFor(modelName)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false);
+        Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false, {
+          id: 'ds.serializer.model-for-key-missing',
+          until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/serializers/rest-serializer.js]: :bomb: `until`

File: packages/ember-data/lib/serializers/rest-serializer.js [hunk 8]
```diff
@@ -1075,13 +1105,19 @@ function _newPushPayload(store, rawPayload) {
...
+  });
   let payload = this.normalizePayload(rawPayload);
 
   for (var prop in payload) {
     var modelName = this.modelNameFromPayloadKey(prop);
     if (!store.modelFactoryFor(modelName)) {
-      Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false);
+      Ember.warn(this.warnMessageNoModelForKey(prop, modelName), false, {
+        id: 'ds.serializer.model-for-key-missing',
+        until: '3.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/serializers/rest-serializer.js]: :bomb: `until`

File: packages/ember-data/lib/system/relationships/belongs-to.js [hunk 9]
```diff
@@ -108,11 +108,25 @@ function belongsTo(modelName, options) {
 
   return computedPolyfill({
     get: function(key) {
-      Ember.warn('You provided a serialize option on the "' + key + '" property in the "' + this._internalModel.modelName + '" class, this belongs in the serializer. See DS.Serializer and it\'s implementations http://emberjs.com/api/data/classes/DS.Serializer.html', !opts.hasOwnProperty('serialize'));
-      Ember.warn('You provided an embedded option on the "' + key + '" property in the "' + this._internalModel.modelName + '" class, this belongs in the serializer. See DS.EmbeddedRecordsMixin http://emberjs.com/api/data/classes/DS.EmbeddedRecordsMixin.html', !opts.hasOwnProperty('embedded'));
+      if (opts.hasOwnProperty('serialize')) {
+        Ember.warn(`You provided a serialize option on the "${key}" property in the "${this._internalModel.modelName}" class, this belongs in the serializer. See DS.Serializer and it's implementations http://emberjs.com/api/data/classes/DS.Serializer.html`, false, {
+          id: 'ds.model.serialize-option-in-belongs-to',
+          until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/system/relationships/belongs-to.js]: :bomb: `until`

File: packages/ember-data/lib/system/relationships/belongs-to.js [hunk 10]
```diff
@@ -108,11 +108,25 @@ function belongsTo(modelName, options) {
...
+        });
+      }
+
+      if (opts.hasOwnProperty('embedded')) {
+        Ember.warn(`You provided an embedded option on the "${key}" property in the "${this._internalModel.modelName}" class, this belongs in the serializer. See DS.EmbeddedRecordsMixin http://emberjs.com/api/data/classes/DS.EmbeddedRecordsMixin.html`, false, {
+          id: 'ds.model.embedded-option-in-belongs-to',
+          until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/system/relationships/belongs-to.js]: :bomb: `until`

File: packages/ember-data/lib/system/relationships/ext.js [hunk 11]
```diff
@@ -261,6 +259,13 @@ Model.reopenClass({
       inverseKind = inverse.kind;
     } else {
       //No inverse was specified manually, we need to use a heuristic to guess one
+      if (propertyMeta.type === propertyMeta.parentType.modelName) {
+        Ember.warn(`Detected a reflexive relationship by the name of '${name}' without an inverse option. Look at http://emberjs.com/guides/models/defining-models/#toc_reflexive-relation for how to explicitly specify inverses.`, false, {
+          id: 'ds.model.reflexive-relationship-without-inverse',
+          until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/system/relationships/ext.js]: :bomb: `until`

File: packages/ember-data/lib/system/relationships/state/relationship.js [hunk 12]
```diff
@@ -209,7 +209,10 @@ Relationship.prototype = {
   },
 
   updateLink: function(link) {
-    Ember.warn("You have pushed a record of type '" + this.record.type.modelName + "' with '" + this.key + "' as a link, but the association is not an async relationship.", this.isAsync);
+    Ember.warn(`You have pushed a record of type '${this.record.type.modelName}' with '${this.key}' as a link, but the association is not an async relationship.`, this.isAsync, {
+      id: 'ds.store.push-link-for-sync-relationship',
+      until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/system/relationships/state/relationship.js]: :bomb: `until`

**Comment by HeroicEric** [comment]
@wecc you're right "Warnings likewise will be required to pass a canonical id:" says nothing about requiring the until

File: packages/ember-data/lib/system/store.js [hunk 13]
```diff
@@ -788,7 +806,10 @@ Store = Service.extend({
           return resolvedRecords.contains(record);
         });
         if (missingRecords.length) {
-          Ember.warn('Ember Data expected to find records with the following ids in the adapter response but they were missing: ' + Ember.inspect(Ember.A(missingRecords).mapBy('id')), false);
+          Ember.warn('Ember Data expected to find records with the following ids in the adapter response but they were missing: ' + Ember.inspect(Ember.A(missingRecords).mapBy('id')), false, {
+            id: 'ds.store.missing-records-from-adapter',
+            until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/system/store.js]: :bomb: `until`

File: packages/ember-data/lib/system/store.js [hunk 14]
```diff
@@ -1891,7 +1935,8 @@ Store = Service.extend({
         })) + ". Make sure they've been defined in your model.",
         filter.call(keysFunc(data), function(key) {
           return !(key === 'id' || key === 'links' || get(type, 'fields').has(key) || key.match(/Type$/));
-        }).length === 0
+        }).length === 0,
+        { id: 'ds.store.unknown-keys-in-payload', until: '2.0.0' }
```
Comment (wecc) [inline packages/ember-data/lib/system/store.js]: :bomb: `until`

File: packages/ember-data/lib/serializers/embedded-records-mixin.js [hunk 15]
```diff
@@ -316,7 +316,11 @@ var EmbeddedRecordsMixin = Ember.Mixin.create({
       key = this.keyForAttribute(attr, 'serialize');
       hasMany = snapshot.hasMany(attr);
 
-      Ember.warn("The embedded relationship '" + key + "' is undefined for '" + snapshot.modelName + "' with id '" + snapshot.id + "'. Please include it in your original payload.", Ember.typeOf(hasMany) !== 'undefined');
+      Ember.warn(
+        `The embedded relationship '${key}' is undefined for '${snapshot.modelName}' with id '${snapshot.id}'. Please include it in your original payload.`,
+        Ember.typeOf(hasMany) !== 'undefined',
+        { id: 'ds.serializer.embedded-relationship-undefined', until: '2.0.0' }
```
Comment (wecc) [inline packages/ember-data/lib/serializers/embedded-records-mixin.js]: :bomb: `until`

File: packages/ember-data/lib/serializers/rest-serializer.js [hunk 16]
```diff
@@ -293,7 +296,10 @@ var RESTSerializer = JSONSerializer.extend({
 
       var typeName = this.modelNameFromPayloadKey(modelName);
       if (!store.modelFactoryFor(typeName)) {
-        Ember.warn(this.warnMessageNoModelForKey(modelName, typeName), false);
+        Ember.warn(this.warnMessageNoModelForKey(modelName, typeName), false, {
+          id: 'ds.serializer.model-for-key-missing',
+          until: '2.0.0'
```
Comment (wecc) [inline packages/ember-data/lib/serializers/rest-serializer.js]: :bomb: `until`

**Comment by HeroicEric** [comment]
All the `until`s have been removed from the warnings

**Comment by wecc** [comment]
@HeroicEric sweet! can you update the message (mentioned above) from `Ember Data 1.0` => `Ember Data 2.0` and I'll be happy to merge!

**Comment by wecc** [comment]
And `[BUGFIX release]` in commit message :)

**Comment by HeroicEric** [comment]
@wecc done

**Comment by wecc** [comment]
Thanks!

**Pull request merged**